projects/packages/sync/src/sync-queue/class-queue-storage-table.php

```diff
@@ -456,4 +476,190 @@ public function delete_items_by_ids( $ids ) {
 			)
 		);
 	}
+
+	/**
+	 * Table initialization
+	 */
+	public static function initialize_custom_sync_table() {
+		/**
+		 * Initialize an instance of the class with a test name, so we can use table prefix and then test if the table is healthy.
+		 */
+		$custom_table_instance = new Queue_Storage_Table( 'test_queue' );
+
+		// Check if the table exists
+		if ( ! $custom_table_instance->custom_table_exists() ) {
+			$custom_table_instance->create_table();
+		}
+
+		if ( ! $custom_table_instance->is_custom_table_healthy() ) {
```

Comment:
`is_custom_table_healthy()` can now return `WP_Error` in addition to bool. This conditional isn't handling the `WP_Error`.

Reply:
Thanks for catching that. Updated the logic.